refactor: move ArtworkMaker to different component

gkartalis opened the pull request:
This PR resolves [] <!-- eg [PROJECT-XXXX] -->

### Description
<!-- Info, implementation, how to get there, before & after screenshots & videos, follow-up work, etc -->

Small refactor that removes the artist name component logic from ArtworkTombstone

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

### Changelog updates

<!-- 📝 Please fill out at least one of these sections. -->
<!-- ⓘ 'User-facing' changes will be published as release notes. -->
<!-- ⌫ Feel free to remove sections that don't apply. -->
<!-- • Write a markdown list or just a single paragraph, but stick to plain text. -->
<!-- 📖 eg. `Enable lotsByFollowedArtists - john` or `Fix phone input misalignment - mary`. -->
<!-- 🤷‍♂️ Replace this entire block with the hashtag `#nochangelog` to avoid updating the changelog. -->

#nochangelog

<!-- end_changelog_updates -->

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

dimatretyak commented on src/app/Scenes/Artwork/Components/ArtworkTombstone.tsx:
```diff
@@ -102,11 +41,7 @@ export const ArtworkTombstone: React.FC<ArtworkTombstoneProps> = ({ artwork }) =
-          : renderMultipleArtists()}
-        {!!(artwork.artists?.length === 0 && artwork.culturalMaker) &&
-          renderArtistName(artwork.culturalMaker, null)}
+        <ArtworkMakerTitle artists={artwork.artists} culturalMaker={artwork.culturalMaker} />
```
**suggestion**: I think we should do something like this

```javascript
<ArtworkMakerTitle artists={artwork.artists}  />
<ArtworkMaker artistName={artwork.culturalMaker} />
```

dimatretyak replied:
aahh, sorry. I misunderstood the prev logic. Sorry to bother you

dimatretyak commented on src/app/Scenes/Artwork/Components/ArtworkMakerTitle.tsx:
```diff
@@ -0,0 +1,103 @@
+    return <ArtworkMultipleMakers artists={artists} />
+  }
+
+  if (artists?.length === 0 && !!culturalMaker) {
+    return <ArtworkMaker artistName={culturalMaker} />
+  }
```
**suggestion**: maybe it would be better to do this check at the very beginning?

dimatretyak commented on src/app/Scenes/Artwork/Components/ArtworkMakerTitle.tsx:
```diff
@@ -0,0 +1,103 @@
+  culturalMaker: ArtworkTombstone_artwork$data["culturalMaker"]
+}
+
+export const ArtworkMakerTitle: React.FC<ComponentProps> = ({ artists, culturalMaker }) => {
```
**suggestion**: use `useFragment` and query all required data in the `ArtworkMakerTitle` component itself

gkartalis replied:
will try it out but not sure if it will work due to the ArtworkScreen being a QueryRenderer

gkartalis replied:
If it doesn't work I can make it with `createFragmentContainer`

dimatretyak commented:
**question**: I think we don't need to [query this data](https://github.com/artsy/eigen/blob/2075db3aba8e73cee66abc11df977fc9b1dbd9fa/src/app/Scenes/Artwork/Components/ArtworkTombstone.tsx#L101) anymore. right?

gkartalis commented:
> **question**: I think we don't need to [query this data](https://github.com/artsy/eigen/blob/2075db3aba8e73cee66abc11df977fc9b1dbd9fa/src/app/Scenes/Artwork/Components/ArtworkTombstone.tsx#L101) anymore. right?

haha, yeah I have removed this in many prs already hahaha removed it in this before you told me also 😂 but also removed it [here](https://github.com/artsy/eigen/pull/7577/files#diff-95775673acb53ce3db09da98651d136829067a3d40b812ab094a3c3aa77a1c0fL166)

dimatretyak commented on src/app/Scenes/Artwork/Components/ArtworkMakerTitle.tsx:
```diff
@@ -0,0 +1,103 @@
+  const { trackEvent } = useTracking()
+
+  const handleArtistTap = (artistHref: string) => {
+    if (!artistHref) {
+      return
+    }
```
**suggestion**: I think it would be better to pass `disabled` prop to `TouchableWithoutFeedback' in this case

dimatretyak commented on src/app/Scenes/Artwork/Components/ArtworkMakerTitle.tsx:
```diff
@@ -0,0 +1,114 @@
+  const { trackEvent } = useTracking()
+
+  const handleArtistTap = (artistHref: string) => {
+    if (!artistHref) {
+      return
+    }
```
this code is no longer needed, since `disabled` prop is passed and `handleArtistTap` will not be called

dimatretyak commented on src/app/Scenes/Artwork/Components/ArtworkTombstone.tsx:
```diff
@@ -102,11 +40,7 @@ export const ArtworkTombstone: React.FC<ArtworkTombstoneProps> = ({ artwork }) =
         </>
       )}
       <Flex flexDirection="row" flexWrap="wrap">
```
**question**: do we need to continue using `Flex` here, since it's already [added here](https://github.com/artsy/eigen/blob/c5555204efc6f86f69a276c0de53b86a0d494dfb/src/app/Scenes/Artwork/Components/ArtworkMakerTitle.tsx#L65)?

dimatretyak commented on src/app/Scenes/Artwork/Components/ArtworkMakerTitle.tests.tsx:
```diff
@@ -0,0 +1,172 @@
+
+jest.unmock("react-relay")
+
+describe("ArtworkMakerTitle", () => {
```
**praise**: thank you for test cases 🙏
